core/client/hdfs/src/test/java/alluxio/hadoop/AbstractFileSystemApiTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.hadoop;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import alluxio.TestLoggerRule;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.URI;
+
+/**
+ * Tests for {@link AbstractFileSystem}. Unlike {@link AbstractFileSystemTest}, these tests only
+ * exercise the public API of {@link AbstractFileSystem}.
+ */
+public final class AbstractFileSystemApiTest {
+
+  @Rule
+  public TestLoggerRule mTestLogger = new TestLoggerRule();
+
+  @After
+  public void after() {
+    HadoopClientTestUtils.resetClient();
+  }
+
+  @Test
+  public void unkownAuthorityTriggersWarning() throws IOException {
+    URI unknown = URI.create("alluxio://test/");
+    FileSystem.get(unknown, new org.apache.hadoop.conf.Configuration());
+    assertTrue(mTestLogger.mAppender.wasLogged("Authority \"test\" is unknown"));
+  }
+
+  @Test
+  public void noAuthorityNoWarning() throws IOException {
```

Comment:
for this case I think it's better not to log a warning. This is the recommended way to write an alluxio uri that picks up its authority from configuration.

Reply:
I don't log a warning in this case